plugins/logistics.cpp

```diff
@@ -358,10 +358,11 @@ class TrainStockProcessor : public StockProcessor {
 
     bool can_designate(color_ostream& out, df::item* item) override {
         auto unit = get_caged_unit(item);
-        return unit &&
-            Units::isTamable(unit) &&
-            !Units::isDomesticated(unit) &&
-            !Units::isMarkedForTraining(unit);
+        return !item->flags.bits.forbid &&
```

Comment:
I suggest moving the forbid check above the call to `get_caged_item` so we don't waste time on caged units if cage is forbidden anyway